Makefile

```diff
@@ -86,16 +86,25 @@ revive: scripts/go/bin/revive
 
 # create docker-compose file with provided sources and start them
 # example: make devenv sources=postgres,openldap
+ifeq ($(sources),)
+devenv:
+	@printf 'You have to define sources for this command \nexample: make devenv sources=postgres,openldap\n'
+else
 devenv: devenv-down
 	$(eval targets := $(shell echo '$(sources)' | tr "," " "))
 
 	@cd devenv; \
-	./create_docker_compose.sh $(targets); \
+	./create_docker_compose.sh $(targets) && rm -f devenv/docker-compose.yaml
+
+	@cd devenv; \
 	docker-compose up -d
+endif
 
 # drop down the envs
 devenv-down:
-	@cd devenv; docker-compose down;
+	@cd devenv; \
+	docker-compose down; \
+	rm -f devenv/docker-compose.yaml
```

Comment:
I don't think we want to remove `docker-compose.yaml` when bringing down the containers. I'd rather prefer the configuration to stay around so that I can re-use it without having to specify the containers the next time I start the env (if my needed datasources are unchanged).

That would require `make devenv` to support not generating the compose file and just running `docker-compose up` when sources arent defined but the compose file already exist.

What do you think about this? (also the rm doesnt work right now because its for the wrong path)

Reply:
> What do you think about this?

I guess my thinking was that reuse will be initiated thro make - you wouldn't use `docker-compose up/down` but `make devenv/devenv-down`.  And if arguments are the same - results should be too. So like pure function situation, no side-effects haha.

And if user would prefer to use docker-compose they always has a possibility to use docker-compose commands and `./create_docker_compose.sh` directly.

> That would require make devenv to support not generating the compose file and just running docker-compose up when sources arent defined but the compose file already exist.

That's the error I was thinking [above](https://github.com/grafana/grafana/pull/17281/files#r287573140)! :))

Reply:
They aren't pure functions though (both have side effects), and that's fine. At the very least I think we should remove the `rm -f docker-compose.yaml`